pkg/command/plugin.go

```diff
@@ -496,16 +502,19 @@ func getInstalledElseAvailablePluginVersion(p *discovery.Discovered) string {
 
 // getInstalledAndMissingContextPlugins returns any context plugins that are not installed
 func getInstalledAndMissingContextPlugins() (installed, missing []discovery.Discovered, err error) {
+	errorList := make([]error, 0)
 	serverPlugins, err := pluginmanager.DiscoverServerPlugins()
 	if err != nil {
-		return nil, nil, err
+		errorList = append(errorList, err)
+		log.Warningf(errorWhileDiscoveringPlugins, err.Error())
```

Comment:
Here we print a warning, but we also eventually return the `errorList` and the calling function also prints a warning for the same error, and it returns the error to Cobra.  So we get the error printed three times.  For example, notice the "404" printed three times below:
```
[!] there was an error while discovering plugins, error information: 'unable to list plugins from discovery source 'default-tmc24': API error, status code: 404'
[!] there was an error while getting installed context plugins, error information: 'unable to list plugins from discovery source 'default-tmc24': API error, status code: 404'
Standalone Plugins
  NAME     DESCRIPTION             TARGET           VERSION          STATUS
  builder  Build Tanzu components  global           v0.90.0-alpha.1  installed
  account  account functionality   mission-control  v9.9.9           installed
  apply    apply functionality     mission-control  v9.9.9           installed

Plugins from Context:  tkg2-1
  NAME                DESCRIPTION                           TARGET      VERSION  STATUS
  cluster             Kubernetes cluster operations         kubernetes  v9.9.9   installed
  feature             Operate on features and featuregates  kubernetes  v9.9.9   installed
  kubernetes-release  Kubernetes release operations         kubernetes  v9.9.9   installed
[x] : unable to list plugins from discovery source 'default-tmc24': API error, status code: 404
```

I suggest we propagate the errors up the call stack, but don't print warnings.  If any function needs to do something special with a particular error, then it can choose to do so and remove the error.  But for right now, we can simply return the errors to Cobra to print to the user.

Reply:
@marckhouzam , I appreciate your suggestions. I will create a separate story to review all CLI commands and address error chaining issues, similar to what we have fixed in this PR. In the new story, I will ensure that we only print the error/warning message once, and it will be done by the actual original caller (command).